torch/serialization.py

```diff
@@ -165,10 +170,15 @@ def _mps_deserialize(obj, location):
     if location == 'mps':
         return obj.mps()
 
+def _meta_deserialize(obj, location):
+    if location == 'meta':
+        return torch.UntypedStorage(obj.nbytes(), device='meta')
```

Comment:
Do we want to allocate the same number of bytes as the prior object since `meta` doesn't actually need any storage ? or is the non-allocation of memory handled somewhere down in the callstack ?

Reply:
We want to match the bytes, since that's how meta tensors normally work (their storages advertise nonzero size but the data pointer is nullptr). Non-allocation is handled by the meta storage allocator